### PR TITLE
Fixed bug where unsetting style properties could mutate the value of other style properties

### PR DESCRIPTION
**Pull request by lucaswoj**
cc @scothis @ansis @jfirebaugh

### CODE REVIEW

File: js/style/style_layer.js [hunk 1]
```diff
@@ -164,18 +164,19 @@ StyleLayer.prototype = {
...
         for (var i = 0; i < removedNames.length; i++) {
-            var spec = this._paintSpecifications[removedNames[i]];
-            applyDeclaration(new StyleDeclaration(spec, spec.default));
+            name = removedNames[i];
+            var spec = this._paintSpecifications[name];
+            applyDeclaration(name, new StyleDeclaration(spec, spec.default));
         }
 
-        function applyDeclaration(declaration) {
+        function applyDeclaration(name, declaration) {
```
Comment (lucaswoj) [inline js/style/style_layer.js]: Not passing `name` as an argument to `applyDeclaration` was a mistake. We captured the outer scope's `name`, which wasn't always the desired value.

Comment (ansis) [inline js/style/style_layer.js]: should we add `block-scoped-var` to the eslint config to catch this kind of bug?

Comment (lucaswoj) [inline js/style/style_layer.js]: That's a good idea. I'll make it a separate PR.

**Comment by ansis** [comment]
looks ready to me!

**Pull request merged**